### ISSUE TITLE
OmniSharp cannot handle multiple csproj files in one directory

### ISSUE THREAD

**Issue by fishfood34** [OmniSharp/omnisharp-vscode#3359]
## Issue Description ##
.sln files have .csproj projects added to them, but instead of running 
```
cd <Path to folder containing .csproj>
dotnet restore <filename>.csproj
```
it appears to be running
```
cd <Path to folder containing .csproj>
dotnet restore
```
## Steps to Reproduce ##
1. Create a directory with more than one .csproj file
2. Create a .sln file that inclues one of those .csproj files
3. Configure omni shart to use that .sln file
4. Open VSCode such that it will run omnisharp using that .sln file

## Expected Behavior ##
OmniSharp runs dotnet restore on the specific .csproj file that is in the .sln file

## Actual Behavior ##
OmniShapr runs dotnet restore in the directory that contains both .csproj files throwing the following error:
```
MSBUILD : error MSB1011: Specify which project or solution file to use because this folder contains more than one project or solution file.
Done: 1.
```

## Environment information ##

**VSCode version**: 1.39.1
**C# Extension**: 1.21.5

<details><summary>Mono Information</summary>
        OmniSharp using built-in mono</details>
<details><summary>Dotnet Information</summary>
.NET Core SDK (reflecting any global.json):
 Version:   2.2.401
 Commit:    729b316c13

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.2.401/

Host (useful for support):
  Version: 2.2.6
  Commit:  7dac9b1b51

.NET Core SDKs installed:
  2.2.401 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-vscode|1.21.5|
|dotenv|mikestead|1.0.1|
|dotnet-test-explorer|formulahendry|0.7.1|
|Go|ms-vscode|0.11.7|
|mssql|ms-mssql|1.6.0|
|python|ms-python|2019.10.41019|
|vscode-docker|ms-azuretools|0.8.1|;

</details>

**Comment by cjvirtucio87** [comment]
Attempted the solution mentioned [here](https://github.com/OmniSharp/omnisharp-vscode/issues/3009#issuecomment-487007332). Still doesn't work:

```
[dbug]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        FAILURE: Could not locate 'c:\Users\cvirtucio\.vscode\extensions\ms-vscode.csharp-1.21.5\.omnisharp\1.32.19-beta.25\
```

I do have a new exception at the end, though:

```
System.InvalidCastException: Specified cast is not valid.
   at Roslyn.Utilities.EventMap.GetRegistries_NoLock[TEventHandler](String eventName)
   at Roslyn.Utilities.EventMap.AddEventHandler[TEventHandler](String eventName, TEventHandler eventHandler)
   at lambda_method(Closure , LifetimeContext , CompositionOperation )
   at System.Composition.Hosting.Core.LifetimeContext.GetOrCreate(Int32 sharingId, CompositionOperation operation, CompositeActivator creator)
   at System.Composition.Hosting.Providers.ImportMany.ImportManyExportDescriptorProvider.<>c__DisplayClass3_2`1.<GetImportManyDescriptor>b__4(ExportDescriptor e)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at lambda_method(Closure , LifetimeContext , CompositionOperation )
   at System.Composition.Hosting.Core.LifetimeContext.GetOrCreate(Int32 sharingId, CompositionOperation operation, CompositeActivator creator)
   at System.Composition.Hosting.Providers.ImportMany.ImportManyExportDescriptorProvider.<>c__DisplayClass3_2`1.<GetImportManyDescriptor>b__4(ExportDescriptor e)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at System.Composition.Hosting.Core.CompositionOperation.Run(LifetimeContext outermostLifetimeContext, CompositeActivator compositionRootActivator)
   at System.Composition.Hosting.Core.LifetimeContext.TryGetExport(CompositionContract contract, Object& export)
   at System.Composition.CompositionContext.GetExport(CompositionContract contract)
   at System.Composition.CompositionContext.GetExports(Type exportType, String contractName)
   at System.Composition.CompositionContext.GetExports[TExport](String contractName)
   at OmniSharp.Stdio.Host.Initialize() in D:\a\1\s\src\OmniSharp.Stdio\Host.cs:line 57
   at OmniSharp.Stdio.Host..ctor(TextReader input, ISharedTextWriter writer, IOmniSharpEnvironment environment, IServiceProvider serviceProvider, CompositionHostBuilder compositionHostBuilder, ILoggerFactory loggerFactory, CancellationTokenSource cancellationTokenSource) in D:\a\1\s\src\OmniSharp.Stdio\Host.cs:line 51
   at OmniSharp.Stdio.Driver.Program.<>c__DisplayClass0_1.<Main>b__1() in D:\a\1\s\src\OmniSharp.Stdio.Driver\Program.cs:line 66
   at OmniSharp.HostHelpers.Start(Func`1 action) in D:\a\1\s\src\OmniSharp.Host\HostHelpers.cs:line 29
[ERROR] Error: OmniSharp server load timed out. Use the 'omnisharp.projectLoadTimeout' setting to override the default delay (one minute).
```

**Comment by geekox86** [comment]
Greetings all. Thank you for wonderful extension.

Any update on this issue?

**Comment by cartermp** [comment]
Closing as fixed by #3617
